test/avocado/selenium-machines-nonroot.py

```diff
@@ -0,0 +1,152 @@
+            self.wait_css('#vm-{}-off'.format(name))
+
+            self.click(self.wait_css('#vm-{}-off-caret'.format(name), cond=clickable))
+            self.click(self.wait_css('#vm-{}-forceOff'.format(name), cond=clickable))
+            self.wait_css('#vm-{}-off'.format(name), cond=invisible)
+            self.wait_css('#vm-{}-run'.format(name))
+
+    def testNoneRootOperationWithVm(self):
+        name = 'staticvm'
+        self.create_vm(name)
+
+        self.machine.execute(
+            'sudo useradd auto && echo "auto" | sudo passwd --stdin auto')
+
+        self.mainframe()
+        self.click(self.wait_css('#navbar-dropdown', cond=clickable))
+        self.click(self.wait_css('#go-logout', cond=clickable))
+
+        self.login('auto', 'auto', authorized=False)
+        self.click(self.wait_link('Virtual Machines', cond=clickable))
+        self.wait_frame("machines")
+
+        self.assertEqual(
+            self.wait_css('#virtual-machines-listing tr td').text,
+            'No VM is running or defined on this host')
+        self.assertEqual(
+            self.wait_css('#app div:nth-child(1) .card-pf-aggregate-status-count').text,
+            '0')
+        self.assertEqual(
+            self.wait_css('#app div:nth-child(2) .card-pf-aggregate-status-count').text,
+            '0')
+
+    def testLibvirtUserOperationWithVm(self):
+        name = 'staticvm'
+        vm_args = self.create_vm(name, graphics='vnc', wait=True)
+
+        self.machine.execute(
+            'sudo useradd -G libvirt testlib && echo "testlib" | sudo passwd --stdin testlib')
+
+        self.mainframe()
+        self.click(self.wait_css('#navbar-dropdown', cond=clickable))
+        self.click(self.wait_css('#go-logout', cond=clickable))
+
+        self.login('testlib', 'testlib', authorized=False)
+        self.click(self.wait_link('Virtual Machines', cond=clickable))
+        self.wait_frame("machines")
+
+        self.wait_css('#vm-{}-row'.format(name))
+        self.click(self.wait_css("tbody tr[data-row-id='vm-{}'] th".format(name), cond=clickable))
+        self.wait_css('#vm-{}-memory'.format(name), cond=text_in, text_='256 MiB')
+        self.wait_css('#vm-{}-vcpus-count'.format(name), cond=text_in, text_='1')
+        self.wait_css('#vm-{}-cputype'.format(name), cond=text_in, text_='custom')
+        self.wait_css('#vm-{}-emulatedmachine'.format(name), cond=text_in, text_='pc')
+        self.wait_css('#vm-{}-bootorder'.format(name), cond=text_in, text_='disk,network')
+
+        self.click(self.wait_css('#vm-{}-pause'.format(name), cond=clickable))
+        self.wait_css('#vm-{}-pause'.format(name), cond=invisible)
+        self.click(self.wait_css('#vm-{}-resume'.format(name), cond=clickable))
+        self.wait_css('#vm-{}-resume'.format(name), cond=invisible)
+        self.click(self.wait_css('#vm-{}-consoles'.format(name), cond=clickable))
+
+        self.operationsForVm(vm_args)
+
+    def testWheelUserOperationWithVm(self):
```

Comment:
Since I see you have a lot of tests with different users performing the same tests. (If not exactly same, similar)
Wouldn't it make sense to create a basic test and then parametrize it to run with all possible users that you have in your suite?

Reply:
I agree with you, but i am afraid the modifications need more time to debug maybe, so, maybe i can add it to my 'todo' list and modify it in next pull request with new cases.

Reply:
Sounds good!